go-controller/pkg/node/gateway_iptables.go

```diff
@@ -467,6 +467,37 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []nodeipt.Rule {
 			Protocol: protocol,
 		},
 	}
+	if util.IsNetworkSegmentationSupportEnabled() {
+		// NOTE: Ordering is important here, the RETURN must come before
+		// the MASQUERADE rule. Please don't change the ordering.
+		srcUDNMasqueradePrefix := config.Gateway.V4MasqueradeSubnet
+		defaultNetworkReservedMasqueradePrefix := config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String() + "/29"
```

Comment:
I know you mention handling masquerade subnet day 2 change later, but it might be good to move the iptrule for the defaultReserved return outside of util.IsNetworkSegmentationSupportEnabled(). It doesn't hurt to have that rule, and it guards against someone changing masquerade subnet, as well as someone disabling segmentation support (do you delete these rules if someone disables segmentation?).

Reply:
we discussed this in team meeting. action items:

1. open a jira card to track disabling the feature
2. do this without a flag so that this always stays as is.